src/types/ajv/Receipts.ts

```diff
@@ -0,0 +1,198 @@
+import { addSchema } from '../../utils/serialization/SchemaHelpers';
+import { AJVSchemaEnum } from '../enum/AJVSchemaEnum';
+// import { schemaAccountsCopy } from './Accounts'; // Import the schema from Accounts.ts
+
+// Define the regex for IPv4 validation (if needed in nested objects)
+const ipv4Regex = /^(?:(?:25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)\.){3}(?:25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)$/;
+
+// Define schemas for nested components
+const schemaProposal = {
+    type: 'object',
+    properties: {
+        applied: { type: 'boolean' },
+        cant_preApply: { type: 'boolean' },
+        accountIDs: { type: 'array', items: { type: 'string' } },
+        beforeStateHashes: { type: 'array', items: { type: 'string' } },
+        afterStateHashes: { type: 'array', items: { type: 'string' } },
+        appReceiptDataHash: { type: 'string' },
+        txid: { type: 'string' }
+    },
+    required: ['applied', 'cant_preApply', 'accountIDs', 'beforeStateHashes', 'afterStateHashes', 'appReceiptDataHash', 'txid'],
+    additionalProperties: false
+};
+
+const schemaSignature = {
+    type: 'object',
+    properties: {
+        owner: { type: 'string' },
+        sig: { type: 'string' }
+    },
+    required: ['owner', 'sig'],
+    additionalProperties: false
+};
+
+const schemaSignedReceipt = {
+    type: 'object',
+    properties: {
+        proposal: schemaProposal,
+        proposalHash: { type: 'string' },
+        signaturePack: {
+            type: 'array',
+            items: schemaSignature
+        },
+        voteOffsets: {
+            type: 'array',
+            items: { type: 'integer' }
+        },
+        sign: { type: 'object', ...schemaSignature },
+        txGroupCycle: { type: 'integer', minimum: 0 }
+    },
+    required: ['proposal', 'proposalHash', 'signaturePack', 'voteOffsets'],
+    additionalProperties: false
+};
+
+const schemaGlobalTxReceipt = {
+    type: 'object',
+    properties: {
+        signs: {
+            type: 'array',
+            items: schemaSignature
+        },
+        tx: {
+            type: 'object',
+            properties: {
+                address: { type: 'string' },
+                addressHash: { type: 'string' },
+                value: {},
```

Comment:
that's because our ajv is aligned to the interface.. and in GlobalTxReceipt.. tx.value : unknown : so in ajv It is equivalent to saying "this can be any type: string, number, object, array, etc."
![image](https://github.com/user-attachments/assets/1ba86fca-e33c-4b00-aa50-26ea700b02ef)

Reply:
so .. {} allows null as value as well ..